src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,159 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { KinveyError } from '../errors';
+import { repositoryProvider } from './repositories';
+import { buildCollectionUrl } from './repositories/utils';
+import {
+  stripTagFromCollectionName,
+  generateEntityId,
+  queryCacheCollectionName,
+  xKivneyRequestStartHeader
+} from './utils';
+
+function getDeltaSetQuery(collectionNameWithTag, query) {
+  const serializedQuery = query ? query.toString() : '';
+
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      const queryCacheQuery = new Query()
+        .equalTo('collectionName', collectionNameWithTag)
+        .and()
+        .equalTo('query', serializedQuery);
+      return offlineRepo.read(queryCacheCollectionName, queryCacheQuery)
+        .then((deltaSetQueryDocs = []) => {
+          if (deltaSetQueryDocs.length > 0) {
+            return deltaSetQueryDocs[0];
+          }
+
+          return {
+            _id: generateEntityId(),
+            collectionName: collectionNameWithTag,
+            query: serializedQuery
+          };
+        });
+    });
+}
+
+function updateDeltaSetQuery(deltaSetQuery, response) {
```

Comment:
This function doesn't seem to need the response. It only needs the header value.